config/config_test.go

```diff
@@ -67,6 +67,97 @@ func TestBlacklist(t *testing.T) {
+		{[]string{"agent", "--password=1234"}, []string{"agent", "--password=********"}},
+		{[]string{"fitz", "-consul_token=1234567890"}, []string{"fitz", "-consul_token=********"}},
+		{[]string{"fitz", "--consul_token=1234567890"}, []string{"fitz", "--consul_token=********"}},
+		{[]string{"fitz", "-consul_token", "1234567890"}, []string{"fitz", "-consul_token", "********"}},
+		{[]string{"fitz", "--consul_token", "1234567890"}, []string{"fitz", "--consul_token", "********"}},
+		{[]string{"python ~/test/run.py --password=1234 -password 1234 -open_password=admin -consul_token 2345 -blocked_from_yaml=1234 &"},
+			[]string{"python", "~/test/run.py", "--password=********", "-password", "********", "-open_password=admin", "-consul_token", "********", "-blocked_from_yaml=********", "&"}},
+		{[]string{"agent", "-PASSWORD", "1234"}, []string{"agent", "-PASSWORD", "********"}},
+		{[]string{"agent", "--PASSword", "1234"}, []string{"agent", "--PASSword", "********"}},
+		{[]string{"agent", "--PaSsWoRd=1234"}, []string{"agent", "--PaSsWoRd=********"}},
+	}
+
+	regexPatterns := setArgsBlacklistPatterns(t)
+
+	for i, _ := range cases {
+		cases[i].cmdline = HideBlacklistedArgs(cases[i].cmdline, regexPatterns)
+		assert.Equal(t, cases[i].parsedCmdline, cases[i].cmdline)
+	}
+}
+
+func TestNoBlacklistedArgs(t *testing.T) {
+	cases := []struct {
+		cmdline       []string
+		parsedCmdline []string
+	}{
+		{[]string{"spidly", "-debug_port=2043"}, []string{"spidly", "-debug_port=2043"}},
+		{[]string{"agent", "start", "-p", "config.cfg"}, []string{"agent", "start", "-p", "config.cfg"}},
+		{[]string{"p1", "-openpassword=admin"}, []string{"p1", "-openpassword=admin"}},
+		{[]string{"p1", "-openpassword", "admin"}, []string{"p1", "-openpassword", "admin"}},
+	}
+
+	regexPatterns := setArgsBlacklistPatterns(t)
+
+	for i, _ := range cases {
+		cases[i].cmdline = HideBlacklistedArgs(cases[i].cmdline, regexPatterns)
+		assert.Equal(t, cases[i].parsedCmdline, cases[i].cmdline)
+	}
+
+}
+
+func BenchmarkRegexMatching1(b *testing.B)       { benchmarckRegexMatching(1, b) }
+func BenchmarkRegexMatching10(b *testing.B)      { benchmarckRegexMatching(10, b) }
+func BenchmarkRegexMatching100(b *testing.B)     { benchmarckRegexMatching(100, b) }
+func BenchmarkRegexMatching1000(b *testing.B)    { benchmarckRegexMatching(1000, b) }
+func BenchmarkRegexMatching10000(b *testing.B)   { benchmarckRegexMatching(10000, b) }
+func BenchmarkRegexMatching100000(b *testing.B)  { benchmarckRegexMatching(100000, b) }
+func BenchmarkRegexMatching1000000(b *testing.B) { benchmarckRegexMatching(1000000, b) }
+
+var avoidOptimization []string
+
+func benchmarckRegexMatching(nbProcesses int, b *testing.B) {
+	runningProcesses := make([][]string, nbProcesses)
+	foolCmdline := []string{"python ~/test/run.py --password=1234 -password 1234 -open_password=admin -consul_token 2345 -blocked_from_yaml=1234 &"}
+
+	for i := 0; i < nbProcesses; i++ {
```

Comment:
I think instead of trying to run this over 1000000 processes, it makes more sense to iterate over `b.N` so we can find out the average cost per regex match call.

Reply:
I've just updated the tests wrapping this part into a iteration over b.N to get an average value for the time that the agent will pass doing the matching on the host. Do you think that it's better to just iterate through the matching function to get the avg per regex call?

Reply:
I think that's find to get the average time matching on the host. I'd kill the 10000 and 100000 benchmarks since they're noise (I'd be surprised to find a host running that many processes at once.

You'll also want to add `b.ResetTimer()` after your setup (and just before the `b.N` loop so that it's not calculated in the benchmarks. (as noted in https://golang.org/pkg/testing/#hdr-Benchmarks)

Reply:
Thanks, Sunny. Just updated the tests with ResetTimer and killed the biggest ones